### ISSUE TITLE
评估结果无法可视化，待解决

### ISSUE THREAD

**Issue by xiaowuzilou** [PaddlePaddle/PaddleSeg#288]
eval.py代码里没有打点，不生成日志文档，无法得到评估结果。希望工程师可以解决一下这个问题

**Comment by wuyefeilin** [comment]
在评估过程中，我们输出了iou的累进结果，评估结束后，会输出整个数据集的平均准确率，平均iou，以及各个类别的准确率，各个类别的iou, 还有kappa指标。

照理说我们已经给出了分割中比较全的评估指标，不知你能否更加详细的描述一下你的问题及需求？

**Comment by xiaowuzilou** [comment]
请问你提到的结果是在哪里输出的呢？命令行里没有输出，也没有生成文档，请问你们的输出结果在哪里可以看到？

**Comment by wuyefeilin** [comment]
命令行里

**Comment by xiaowuzilou** [comment]
(base) E:\PaddleSeg-develop>python pdseg/eval.py --use_gpu --cfg ./configs/unet_optic.yaml
{'AUG': {'AUG_METHOD': 'unpadding',
         'FIX_RESIZE_SIZE': (512, 512),
         'FLIP': False,
         'FLIP_RATIO': 0.5,
         'INF_RESIZE_VALUE': 500,
         'MAX_RESIZE_VALUE': 600,
         'MAX_SCALE_FACTOR': 2.0,
         'MIN_RESIZE_VALUE': 400,
         'MIN_SCALE_FACTOR': 0.5,
         'MIRROR': True,
         'RICH_CROP': {'ASPECT_RATIO': 0.33,
                       'BLUR': False,
                       'BLUR_RATIO': 0.1,
                       'BRIGHTNESS_JITTER_RATIO': 0.5,
                       'CONTRAST_JITTER_RATIO': 0.5,
                       'ENABLE': False,
                       'MAX_ROTATION': 15,
                       'MIN_AREA_RATIO': 0.5,
                       'SATURATION_JITTER_RATIO': 0.5},
         'SCALE_STEP_SIZE': 0.25},
 'BATCH_SIZE': 4,
 'DATALOADER': {'BUF_SIZE': 256, 'NUM_WORKERS': 8},
 'DATASET': {'DATA_DIM': 3,
             'DATA_DIR': './dataset/optic_disc_seg/',
             'IGNORE_INDEX': 255,
             'IMAGE_TYPE': 'rgb',
             'NUM_CLASSES': 2,
             'PADDING_VALUE': [127.5, 127.5, 127.5],
             'SEPARATOR': ' ',
             'TEST_FILE_LIST': './dataset/optic_disc_seg/test_list.txt',
             'TEST_TOTAL_IMAGES': 180,
             'TRAIN_FILE_LIST': './dataset/optic_disc_seg/train_list.txt',
             'TRAIN_TOTAL_IMAGES': 900,
             'VAL_FILE_LIST': './dataset/optic_disc_seg/val_list.txt',
             'VAL_TOTAL_IMAGES': 180,
             'VIS_FILE_LIST': './dataset/optic_disc_seg/test_list.txt'},
 'EVAL_CROP_SIZE': (512, 512),
 'FREEZE': {'MODEL_FILENAME': '__model__',
            'PARAMS_FILENAME': '__params__',
            'SAVE_DIR': 'freeze_model'},
 'MEAN': [0.5, 0.5, 0.5],
 'MODEL': {'BN_MOMENTUM': 0.99,
           'DEEPLAB': {'ASPP_WITH_SEP_CONV': True,
                       'BACKBONE': 'xception_65',
                       'DECODER_USE_SEP_CONV': True,
                       'DEPTH_MULTIPLIER': 1.0,
                       'ENABLE_DECODER': True,
                       'ENCODER_WITH_ASPP': True,
                       'OUTPUT_STRIDE': 16},
           'DEFAULT_EPSILON': 1e-05,
           'DEFAULT_GROUP_NUMBER': 32,
           'DEFAULT_NORM_TYPE': 'bn',
           'FP16': False,
           'HRNET': {'STAGE2': {'NUM_CHANNELS': [40, 80], 'NUM_MODULES': 1},
                     'STAGE3': {'NUM_CHANNELS': [40, 80, 160],
                                'NUM_MODULES': 4},
                     'STAGE4': {'NUM_CHANNELS': [40, 80, 160, 320],
                                'NUM_MODULES': 3}},
           'ICNET': {'DEPTH_MULTIPLIER': 0.5, 'LAYERS': 50},
           'MODEL_NAME': 'unet',
           'MULTI_LOSS_WEIGHT': [1.0],
           'PSPNET': {'DEPTH_MULTIPLIER': 1, 'LAYERS': 50},
           'SCALE_LOSS': 'DYNAMIC',
           'UNET': {'UPSAMPLE_MODE': 'bilinear'}},
 'NUM_TRAINERS': 1,
 'SLIM': {'KNOWLEDGE_DISTILL': False,
          'KNOWLEDGE_DISTILL_IS_TEACHER': False,
          'KNOWLEDGE_DISTILL_TEACHER_MODEL_DIR': '',
          'NAS_ADDRESS': '',
          'NAS_IS_SERVER': True,
          'NAS_PORT': 23333,
          'NAS_SEARCH_STEPS': 100,
          'NAS_SPACE_NAME': '',
          'NAS_START_EVAL_EPOCH': 0,
          'PREPROCESS': False,
          'PRUNE_PARAMS': '',
          'PRUNE_RATIOS': []},
 'SOLVER': {'BEGIN_EPOCH': 1,
            'CROSS_ENTROPY_WEIGHT': None,
            'DECAY_EPOCH': [10, 20],
            'GAMMA': 0.1,
            'LOSS': ['softmax_loss'],
            'LOSS_WEIGHT': {'BCE_LOSS': 1,
                            'DICE_LOSS': 1,
                            'LOVASZ_HINGE_LOSS': 1,
                            'LOVASZ_SOFTMAX_LOSS': 1,
                            'SOFTMAX_LOSS': 1},
            'LR': 0.001,
            'LR_POLICY': 'poly',
            'LR_WARMUP': False,
            'LR_WARMUP_STEPS': 2000,
            'MOMENTUM': 0.9,
            'MOMENTUM2': 0.999,
            'NUM_EPOCHS': 10,
            'OPTIMIZER': 'adam',
            'POWER': 0.9,
            'WEIGHT_DECAY': 4e-05},
 'STD': [0.5, 0.5, 0.5],
 'TEST': {'TEST_MODEL': './saved_model/unet_optic/final'},
 'TRAIN': {'MODEL_SAVE_DIR': './saved_model/unet_optic/',
           'PRETRAINED_MODEL_DIR': './pretrained_model/unet_bn_coco/',
           'RESUME_MODEL_DIR': '',
           'SNAPSHOT_EPOCH': 5,
           'SYNC_BATCH_NORM': False},
 'TRAINER_ID': 0,
 'TRAIN_CROP_SIZE': (512, 512)}
#Device count: 1
W0601 23:32:51.594745 24660 device_context.cc:252] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 10.1, Runtime API Version: 10.0
W0601 23:32:51.633760 24660 device_context.cc:260] device: 0, cuDNN Version: 7.3.
load test model: ./saved_model/unet_optic/final

**Comment by xiaowuzilou** [comment]
上面是评估的运行结果，你说的是在这个命令行里看嘛？没有看到评估结果呢。。

**Comment by wuyefeilin** [comment]
![image](https://user-images.githubusercontent.com/30919197/84113716-ad09c300-aa5d-11ea-847e-438d8f923969.png)
没有这个吗

**Comment by xiaowuzilou** [comment]
deeplabv3+和u-net都没有

**Comment by xiaowuzilou** [comment]
模型训练完成，使用下述命令启动评估

python pdseg/eval.py --use_gpu --cfg ./configs/deeplabv3p_xception65_optic.yaml
请问你发布的图片是在这行命令下面出来的嘛？如果是，我在本地跑的出来的是我粘贴的，没有出来你图片里的

**Comment by wuyefeilin** [comment]
@xiaowuzilou 
是的，你那边有报错吗？ 还是根本没跑起来？，

**Comment by xiaowuzilou** [comment]
不报错，后面的可视化也能得到结果，就是不出评估结果

**Comment by wuyefeilin** [comment]
程序有在运行吗？还是根本没有进行评估？我这边不能复现你的结果，没办法排查原因。你也可以把你的模型，配置，数据，运行命令打包一份给我，我看看能不能复现

**Comment by xiaowuzilou** [comment]
程序在服务器上跑的，评估结果就是上面我粘贴的那样，deeplabv3+和unet都是那种形式，没有出现你截图里的结果。最后可以得到测试集的可视化结果，这算不算可以复现程序

**Comment by wuyefeilin** [comment]
![image](https://user-images.githubusercontent.com/30919197/84119060-71272b80-aa66-11ea-825a-305f4e1f267e.png)
输出到这一行程序直接卡住了吗？还是运行结束了。 你也可以进入eval.py代码里面调试看看，是哪边卡住了

**Comment by xiaowuzilou** [comment]
需要源码编译是吗

**Comment by wuyefeilin** [comment]
不需要，直接就是python代码，你可以在代码中间插入print语句打印一些信息出来看看哪边出问题了
